### PR TITLE
querydb: make android query work for kotlin & java

### PR DESCRIPTION
**Pull request by claudiuvursache**
also adds some more primitive type inference using imports

### CODE REVIEW

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/KtPsiToAst.scala [hunk 1]
```diff
@@ -1610,8 +1610,18 @@ trait KtPsiToAst {
     argIdx: Option[Int],
     argName: Option[String] = None
   )(implicit typeInfoProvider: TypeInfoProvider): Ast = {
-    val typeFullName = registerType(typeInfoProvider.typeFullName(expr, TypeConstants.any))
-    val name         = expr.getIdentifier.getText
+    val fallBackTypeName = scope.lookupVariable(expr.getIdentifier.getText) match {
```
Comment (claudiuvursache) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/KtPsiToAst.scala:1613]: ugly as hell, i know. will need a proper pattern here and in similar places

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala [hunk 2]
```diff
@@ -38,4 +38,40 @@ class IdentifierTests extends KotlinCode2CpgFixture(withOssDataflow = false) {
       cpg.identifier("args").columnNumber.l.head shouldBe 18
     }
   }
+
+  "CPG for code with various missing types, but some information available in the imports" should {
+    val cpg = code("""|package io.vrooom.vulnerableapp
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala:43]: You may remove the leading `|`.

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala [hunk 3]
```diff
@@ -38,4 +38,40 @@ class IdentifierTests extends KotlinCode2CpgFixture(withOssDataflow = false) {
...
+       |
+       |import android.os.Bundle
+       |import android.support.v7.app.AppCompatActivity
+       |import android.webkit.WebView
+       |import android.content.IntentFilter
+
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala:49]: And here is one `|` missing.

**Review by max-leuthaeuser — APPROVED** [review]

**Review by johannescoetzee — APPROVED** [review]

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala (hunk 2 above)
Comment (claudiuvursache) [reply joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala]: thx

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala (hunk 3 above)
Comment (claudiuvursache) [reply joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/IdentifierTests.scala]: included, danke

**Pull request merged**